### ISSUE TITLE
Please consider distributing flatpak as a flatpak

### ISSUE THREAD

**Issue by pvaret** [flatpak/flatpak#3159]
Currently flatpak users are dependant on their distro updating the flatpak package. For instance the newly released flatpak 1.5, which fixes a few painful issues, is not in Ubuntu and will not be for a while.

If flatpak itself was available as a flatpak package, we could still install it using the distro's provided flatpak as bootstrap, and then we'd be independent of the distro's release schedule and/or lack of motivation to keep up with flatpak releases.

Is that technically feasible? Is it desirable to other people than me?

**Comment by aiamuzz** [comment]
hmmmmm ... i don't think that would be possible ... all flatpak apps work bcos the flatpak package is installed on a distribution natively ... i am just wondering what will enable the running of the flatpak'd flatpak app !!!

**Comment by matthiasclasen** [comment]
it would be elegant, from some perspective, but I'm afraid that it does not easily fit into the flatpak model of doing things

**Comment by aiamuzz** [comment]
> it would be elegant, from some perspective, but I'm afraid that it does not easily fit into the flatpak model of doing things

@matthiasclasen ...
oh its possible ... sure it would be elegant ?

i wonder why its not being considered ... any thread where this option has been discussed in detail ?

**Comment by matthiasclasen** [comment]
I didn't say it is possible - I'm pretty sure it isn't.  Flatpak's promise is that the things it downloads (apps) are run in a confined environment (the sandbox). But flatpak itself needs to run unconfined on the host in order to be able to set up the sandboxes.

So, if you want to make flatpak download itself, you need to break that promise (or make sandbox setup possible from inside a sandbox, which requires kernel-level changes that are unlikely to ever happen).

**Comment by pvaret** [comment]
Ah, darn. I was hoping there existed some escape hatch to allow programs
such as Flatpak to ship on Flatpak.

Then, back to my earlier question of making Flatpak releases
distro-independent. Would shipping flatpak itself as an AppImage work? I
don't believe they're sandboxed.

**Comment by aiamuzz** [comment]
> I think shipping flatpak as an appimage (even if it could be made to work) would be too much irony for us to handle.

irony ... true ... taking the support of your competitor in the field to launch you higher !!!



> If there is any particular fix that fixes a particularly painful issue, you could ask for that to be backported (either here, or in the distro package).

Oh ... i didn't know we could request for that ... my distro Linux Deepin is painfully primitive ... i think they are on `Version 0.11` was painful ... until i came to know of the stretch backports option ... thankfully i am now on `stretch-backports,now 1.2.4-1~bpo9+1 amd64`

**Comment by pvaret** [comment]
Yeah, I won't lie, I do like the idea of competing package solutions
supporting each other where appropriate. Here's to a better world. :)

I wasn't aware that 1.5 is a development release. That's a bummer. That
said, it looks like the issues that blocked me last time I tried out
Flatpak may have been fixed and the fix backported.

My use case is to automate updates and cleanup, which was blocked by #3004
and an issue where package org.freedesktop.Platform.openh264 failed to
update because it "needs a later flatpak version". But neither of these
problems seem to be occurring anymore. Therefore it looks like I do in fact
not need to install Flatpak 1.5!

This ticket can remain closed. Thank you for the good work. :)
